File: my-react-app/src/types/product.ts
export interface ProductIngredient {
  ingredientId: number;
  ingredientName?: string;
  amountNeeded: number;
  unit?: string;
}

export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  category: string;
  image: string;
  ingredients: string; // Legacy
  dietary: string;     // Legacy
  productIngredients: ProductIngredient[];
}

/** DTO for create/update — same fields minus id */
export interface ProductFormDto {
  name: string;
  description: string;
  price: number;
  category: string;
  image: string;
  ingredients: string; 
  dietary: string;     
  productIngredients: ProductIngredient[];
}

File: my-fullstack-app/server/Models/Product.cs
namespace server.Models;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = "";
    public decimal Price { get; set; }
    public string Description { get; set; } = "";
    public string Category { get; set; } = "Menu";
    public string Image { get; set; } = "";
    /// <summary>JSON array string, e.g. ["eggs","bacon","butter"]</summary>
    
     public string Ingredients { get; set; } = "[]";

    /// <summary>JSON array string, e.g. ["vegetarian","gluten-free"]</summary>
    public string Dietary { get; set; } = "[]";

    public ICollection<ProductIngredient> ProductIngredients { get; set; } = new List<ProductIngredient>();
}
